src/Data/Entity.php

```diff
@@ -3,52 +3,60 @@
 
 abstract class Entity implements \JsonSerializable
 {
-    protected $properties = [];
-
     protected $id;
+    protected $createdAt;
+    protected $updatedAt;
+    protected $deleted = false;
 
-    private $data;
-
-    private $callMethods = ['get', 'set'];
+    public function getId()
+    {
+        return $this->id;
+    }
 
-    public function __construct($data = [])
+    public function setId($id)
     {
-        $this->data = $data;
+        $this->id = (int)$id;
     }
 
-    public function getId()
+    public function getCreatedAt()
```

Comment:
Along with being immutable, I don't see a strong reason to have `get` prefixes (because there are no `set` prefixes):

```php
public function createdAt()
{
    return $this->createdAt;
}
```